stackslib/src/burnchains/mod.rs

```diff
@@ -473,6 +510,15 @@ impl PoxConstants {
         }
     }
 
+    /// What's the first block in the prepare phase
+    pub fn prepare_phase_start(&self, first_block_height: u64, reward_cycle: u64) -> u64 {
+        let reward_cycle_start =
+            self.reward_cycle_to_block_height(first_block_height, reward_cycle);
```

Comment:
I know that this should never underflow as the length of the reward cycle and prepare length are known. Do we still rather have an explicit check with a panic if self.prepare_length > reward_cycle_start + self.reward_cycle_length since this does introduce a potential underflow ?

Reply:
This is checked in `PoxConstants::new()` :+1: